### DISCUSSION TITLE
Rename file

### DISCUSSION THREAD

**Discussion by agingorange** [twpayne/chezmoi#2143] [Q&A] [answered]
What is the "chezmoi way" of renaming a file?

**Comment by twpayne** [comment, marked as answer]
As renaming a file can only happen once, I'd suggest an idempotent run script, something like `run_rename.sh`:

```sh
#!/bin/sh

if [ -e $OLDNAME ] ; then
    mv $OLDNAME $NEWNAME
fi
```

**Reply by agingorange** [reply]
Thank you.

**Reply by alper** [reply]
> can only happen once

What does this mean? Why can it only happen once?

**Reply by Lockszmith-GH** [reply]
Once the file has been renamed, you will not rename it again - hence, it will only happen once.

This is, if you want to preserve the renaming across other locations.
There are multiple thread on the topic of synchronizing rename and deletions, search around and ask a new question if you need more help. I've even posted an FR about it: #4361
